diesel/src/migrations/mod.rs

```diff
@@ -108,6 +108,34 @@ pub fn run_pending_migrations_in_directory<Conn>(conn: &Conn, migrations_dir: &P
     run_migrations(conn, all_migrations, output)
 }
 
+// Returns true if there are outstanding migrations in the migrations directory, otherwise
+// returns false. Returns an `Err` if there are problems with migration setup.
+///
+/// See the [module level documentation](index.html) for information on how migrations should be
+/// structured, and where Diesel will look for them by default.
+pub fn any_pending_migrations<Conn>(conn: &Conn) -> Result<bool, RunMigrationsError> where
+    Conn: MigrationConnection
+{
+    let migrations_dir = find_migrations_directory()?;
+    let pending_migrations = pending_migrations_in_directory(conn, &migrations_dir)?;
+
+    Ok(!pending_migrations.is_empty())
```

Comment:
Your code is nice, I think you can write it a bit more concisely, though. What you are checking is that there is any migration that was not run, right? You don't ever need the list of pending migrations. So, let's use `.any`:

```rust
let all_migrations = migrations_in_directory(migrations_dir)?;
let already_run = conn.previously_run_migration_versions()?;

Ok(all_migrations.into_iter()
    .any(|m| !already_run.contains(&m.version().to_string()))
```